### DISCUSSION TITLE
Limit on number of Options in Option Picker

### DISCUSSION THREAD

**Discussion by ndlang** [Budibase/budibase#6225] [Help] [answered]
Is there a Limit on number of Options in Option Picker? I am populating an op picker with data from a mysql table and not all the options are present.

Data in the Data Provider:
![image](https://user-images.githubusercontent.com/17033357/172393160-839f4f4f-8255-4084-8d8c-9da88bcd0aa9.png)

Options available:
![image](https://user-images.githubusercontent.com/17033357/172393541-8afb24da-266f-4e9a-97cc-772c903b1398.png)

Component config:
![image](https://user-images.githubusercontent.com/17033357/172393907-f23b8b8a-bf2f-49a6-a0dc-b72fff414865.png)

Total number of Orgs = 379

**Comment by melohagan** [comment, marked as answer]
Hi @ndlang 

If you scroll through the company ids, can you find Mishawaka?

I just want to determine if the issue is with the options or the search functionality.

**Reply by ndlang** [reply]
Hey thanks for the reply. I only get to the C's:
![image](https://user-images.githubusercontent.com/17033357/172616196-e5b1748d-3589-4da6-8414-4f3014e12f3a.png)

Which is 47 records out of 379.

**Reply by melohagan** [reply]
Ah I think I see what's happening now. When you use a data provider as the source for an Options Picker, it also adheres to the limit on the data provider being used.
E.g.
![Screenshot 2022-06-08 at 14 14 27](https://user-images.githubusercontent.com/101575380/172625415-f801a27d-c272-4f46-8a6f-2878c2151445.png)
![Screenshot 2022-06-08 at 14 14 53](https://user-images.githubusercontent.com/101575380/172625509-1c73c83f-09bb-43d9-9246-2b8c75e9cbeb.png)

**Reply by melohagan** [reply]
Looking through the code it appears the default limit is 50, and the max may be 200.

**Reply by hawkinsjon** [reply]
Is there a way to increase the max limit of 200 for the option picker? I'm in a situation where ideally allow 14,000 records on the option picker (with a autocomplete). I guess another solution would be to have a search box w/ autocomplete although I don't see that component in the list.

**Reply by melohagan** [reply]
Hey @hawkinsjon There's no setting for changing that. Presumably thousands of options on screen could be a bit janky.
Custom components are going to be released this month - so what you're looking for seems like a good candidate for a new component, maybe a "Search Picker" that lists options only from typed input.
